guide/guide/.js/src/main/scala/io/udash/web/guide/demos/AutoDemo.scala

```diff
@@ -3,16 +3,20 @@ package io.udash.web.guide.demos
 import com.avsystem.commons.SharedExtensions
 import io.udash.web.commons.components.CodeBlock
 import io.udash.web.guide.styles.partials.GuideStyles
+import scalatags.JsDom.all._
 
 trait AutoDemo extends SharedExtensions {
 
-  import scalatags.JsDom.all._
 
   final def demoWithSnippet(): (Modifier, Modifier) = {
     val (demo, code) = demoWithSource()
-    (demo, CodeBlock.lines(code.drop(1).map(_.drop(2)))(GuideStyles))
+    (demo, AutoDemo.snippet(code))
   }
 
-  protected def demoWithSource(): (Modifier, Iterator[String])
+  protected def demoWithSource(): (Modifier, String)
 
 }
+
+object AutoDemo {
+  def snippet(code: String): Modifier = CodeBlock.lines(code.linesIterator.drop(1).map(_.drop(2)).toList.view.dropRight(1).iterator)(GuideStyles)
```

Comment:
Maybe we could add `from` and `to` params and just provide subset of lines(initial idea but you get the point), to provide possibility of migrating source codes dependent of another source codes?

Reply:
1. `dropRight` is deduplicated from callsites, where it was required to remove the final curly brace. I think it's here to stay and unrelated to your second q
2. Do we want to though? I'm pretty sure the lines will move (e.g. via automatic formatting of a file)  and it will require a lot of care to keep this consistent, basically as bad as before this PR.